### DISCUSSION TITLE
running runGenie3 using a custom list of TFs

### DISCUSSION THREAD

**Discussion by arijita88** [aertslab/SCENIC#252] [R] [answered]
Hi,
      How can I run runGenie3 using a custom list of transcription factors(TFs)? For example, as mentioned in the GENIE3 vignette, it can be done using regulators <- c("Gene1", "Gene2", "Gene3") and then weightMat <- GENIE3(exprMatr, regulators=regulators). My .loom file is huge and runGenie3 detects 1286 TFs in it. I need to run the .loom file just for 10 TFs. Any advice on this will be really helpful. Thank you.

**Comment by jonesphil93** [comment]
Hi!

Did you figure out how to solve this problem? I would also like to use some TFs as regulators, but haven't found a solution to this problem yet...
Thanks!

**Comment by arijita88** [comment]
Hi,
     Sorry I haven't found a solution to this problem yet. I really want to
figure out how to do this. Please let me know if you have any leads on this.

Thank You
Arijita

On Thu, Oct 22, 2020 at 7:18 AM jonesphil93 <notifications@github.com>
wrote:

> Hi!
>
> Did you figure out how to solve this problem? I would also like to use
> some TFs as regulators, but haven't found a solution to this problem yet...
> Thanks!
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/aertslab/SCENIC/issues/114#issuecomment-714526567>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AIGZZBALIOZ2ZHJ6LABXEHLSMA5KPANCNFSM4KEAVOCA>
> .
>


-- 
*************************************************************
Arijita Sarkar
University of Southern California
Department of Orthopaedic Surgery
Stem Cell Research and Regenerative Medicine
Evseenko Lab
1450 Biggy St. NRT4509
Los Angeles, CA 90033
arijitas@usc.edu
**************************************************************

**Comment by s-aibar** [comment, marked as answer]
Dear @arijita88 and @jonesphil93 ,

I have added the option to select the input TFs to the `runGenie3` function (argument `allTFs`). 

However, beware that using only a small subset of TFs will modify the results of the pipeline! Keep in mind that this isn't a supported modification, so you should probably test it yourself: for example, keep at least a few extra TFs as negative controls, and also check that the thresholds for creating the co-expression modules still make sense (i.e. manually run the vignette `detailedStep_1_coexNetwork2modules.Rmd`, to confirm everything is fine... )

**Comment by jonesphil93** [comment]
Dear @s-aibar ,

thank you very much for your help and efforts!

Sorry for being so stupid, but you provide an example how to use the allTFs argument?
E.g., if I want to run it for example only for the TFs Id2 and Prdm1 (because they are highly expressed in my dataset) and use Bcl6 as negative control, what will my argument allTFs look like?

Thank you very much!

**Comment by s-aibar** [comment]
Just provide them all as a vector:

e.g.: 
`allTFs=c("Id2", "Prdm1", "Bcl6")`

**Comment by jonesphil93** [comment]
Thanks!

**Comment by arijita88** [comment]
Hi @s-aibar <https://github.com/s-aibar>,
                      Thank you so much for helping us with this.

On Fri, Oct 23, 2020 at 6:18 AM jonesphil93 <notifications@github.com>
wrote:

> Thanks!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/aertslab/SCENIC/issues/114#issuecomment-715335361>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AIGZZBEMLB4QQXOYXMNVWQLSMF7DNANCNFSM4KEAVOCA>
> .
>


-- 
*************************************************************
Arijita Sarkar
University of Southern California
Department of Orthopaedic Surgery
Stem Cell Research and Regenerative Medicine
Evseenko Lab
1450 Biggy St. NRT4509
Los Angeles, CA 90033
arijitas@usc.edu
**************************************************************
